### PR TITLE
[slack] allow the user to choose the message format to send

### LINKED ISSUE #3226: [slack] Remove previews
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature, Other

### 🔖 Feature description

image

I had someone ask me for this, and I am not sure if it is possible.  But the slack integration adds a preview with what looks like very similar information from the message.  This appears a bit superfluous and it would be nice to be able to disable this.

### ✔️ Solution

Add an option to the web gui to disable link and media unfurling.
https://api.slack.com/reference/messaging/link-unfurling#no_unfurling_please

Not really sure where the preview is coming from, it's not really media and not really a link.  But this should prevent this from happening in the slack API/Webhook call:

```json
{
    "text": "Check out the <https://api.slack.com|Slack API>",
    "unfurl_links": false,
    "unfurl_media": false
}
```

So a simple slider for "Disable Preview", "Disable Link Unfurling" in the slack integration edit menu with the other sliders should be functional and still look aesthetically pleasing.  You could also do a single slider "No previews" or something instead.

image

### ❓ Alternatives

Not sure of any.

### 📝 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by DaanMeijer**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Fixes #3226

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- User interface (UI)
- New feature (non-breaking change which adds functionality)
- Breaking change (a fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)
Non-richt text message:
image

Rich text message:
image

### CODE REVIEW

File: src/components/notifications/Slack.vue [hunk 1]
```diff
@@ -9,6 +9,12 @@
         <label for="slack-channel" class="form-label">{{ $t("Channel Name") }}</label>
         <input id="slack-channel-name" v-model="$parent.notification.slackchannel" type="text" class="form-control">
 
+        <label class="form-label">{{ $t("Message format") }}</label>
+        <div class="form-check form-switch">
+            <input id="slack-text-message" v-model="$parent.notification.slackrichmessage" type="checkbox" class="form-check-input">
+            <label for="slack-text-message" class="form-label">{{ $t("Send rich messages") }}</label>
```
Comment (CommanderStorm) [inline src/components/notifications/Slack.vue:15]: Please add the new translaton keys `Message format` and `Send rich messages` to [`en.json`](https://github.com/louislam/uptime-kuma/blob/master/src/lang/en.json).

Without these keys being there, our translators won't be able to translate this string in weblate ^^

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Thanks, mostly mergable.
I have noted one thing regarding translation keys below.

File: src/components/notifications/Slack.vue (hunk 1 above)
Comment (DaanMeijer) [reply src/components/notifications/Slack.vue:15]: Done!

**Comment by DaanMeijer** [comment]
@CommanderStorm To be sure: is any more action required from me?

**Review by CommanderStorm — APPROVED** [review]
Thanks for the enhancement to the notification provider! 🎉 
I have changed this to v2.0, as this is more of a feature than a bugfix.



> [!NOTE]
> This PR is part of the [`v2.0` merge window](https://github.com/louislam/uptime-kuma/milestone/24) => see #4500 for the bugs that need to be addressed before we can ship this release ^^

<sub>All changes in this PR are small and uncontroversial ⇒ merging with junior maintainer approval</sub>

**Comment by pelligrag** [comment]
Rich messages are not functioning on version 2.0 beta. I don't receive any notifications if I enable rich messages in Slack notifications.

**Comment by DaanMeijer** [comment]
> Rich messages are not functioning on version 2.0 beta. I don't receive any notifications if I enable rich messages in Slack notifications.

This is fixed in https://github.com/louislam/uptime-kuma/pull/5196#issuecomment-2416740399

**Pull request merged**